bottlecap/src/config/mod.rs

```diff
@@ -453,90 +450,20 @@ impl Default for Config {
             appsec_waf_timeout: Duration::from_millis(5),
             api_security_enabled: true,
             api_security_sample_delay: Duration::from_secs(30),
-
-            extension_version: None,
         }
     }
 }
 
-fn log_fallback_reason(reason: &str) {
-    println!("{{\"DD_EXTENSION_FALLBACK_REASON\":\"{reason}\"}}");
-}
-
-fn fallback(config: &Config) -> Result<(), ConfigError> {
-    // Customer explicitly opted out of the Next Gen extension
-    let opted_out = match config.extension_version.as_deref() {
-        Some("compatibility") => true,
-        // We want customers using the `next` to not be affected
-        _ => false,
-    };
-
-    if opted_out {
-        log_fallback_reason("extension_version");
-        return Err(ConfigError::UnsupportedField(
-            "extension_version".to_string(),
-        ));
-    }
-
-    // ASM / .NET
-    // todo(duncanista): Remove once the .NET runtime is fixed
-    if config.serverless_appsec_enabled && has_dotnet_binary() {
-        log_fallback_reason("serverless_appsec_enabled_dotnet");
-        return Err(ConfigError::UnsupportedField(
-            "serverless_appsec_enabled_dotnet".to_string(),
-        ));
-    }
-
-    // OTLP
-    let has_otlp_config = config
-        .otlp_config_receiver_protocols_grpc_endpoint
-        .is_some()
-        || config
-            .otlp_config_receiver_protocols_grpc_transport
-            .is_some()
-        || config
-            .otlp_config_receiver_protocols_grpc_max_recv_msg_size_mib
-            .is_some()
-        || config.otlp_config_metrics_enabled
-        || config.otlp_config_metrics_resource_attributes_as_tags
-        || config.otlp_config_metrics_instrumentation_scope_metadata_as_tags
-        || config.otlp_config_metrics_tag_cardinality.is_some()
-        || config.otlp_config_metrics_delta_ttl.is_some()
-        || config.otlp_config_metrics_histograms_mode.is_some()
-        || config.otlp_config_metrics_histograms_send_count_sum_metrics
-        || config.otlp_config_metrics_histograms_send_aggregation_metrics
-        || config
-            .otlp_config_metrics_sums_cumulative_monotonic_mode
-            .is_some()
-        || config
-            .otlp_config_metrics_sums_initial_cumulativ_monotonic_value
-            .is_some()
-        || config.otlp_config_metrics_summaries_mode.is_some()
-        || config
-            .otlp_config_traces_probabilistic_sampler_sampling_percentage
-            .is_some()
-        || config.otlp_config_logs_enabled;
```

Comment:
We still don't support any of this, should we warn at least? It has never been supported, but just wondering

Reply:
I think it's ok for now.